examples/platforms/utils/mac_frame.cpp

```diff
@@ -131,16 +132,11 @@ bool otMacFrameIsVersion2015(const otRadioFrame *aFrame)
     return static_cast<const Mac::Frame *>(aFrame)->IsVersion2015();
 }
 
-otError otMacFrameGenerateImmAck(const otRadioFrame *aFrame, bool aIsFramePending, otRadioFrame *aAckFrame)
+void otMacFrameGenerateImmAck(const otRadioFrame *aFrame, bool aIsFramePending, otRadioFrame *aAckFrame)
 {
-    otError error = OT_ERROR_NONE;
-
-    VerifyOrExit(aFrame != NULL && aAckFrame != NULL, error = OT_ERROR_INVALID_ARGS);
+    assert(aFrame != NULL && aAckFrame != NULL);
```

Comment:
use `OT_ASSERT`

Reply:
I'm not sure about this. As I know, `OT_ASSERT` is used In `src` directory and `assert` is used in `examples` directory. And currently only `assert` is used in `examples` directory. Shall we use `OT_ASSERT` here?

Reply:
@abtink any reason for not applying OT_ASSERT to examples in #4665?

Reply:
`debug.hpp` is a core-internal private header file, so technically should not be exposed/seen by the platform code. And `OT_ASSERT` macro is intended for the OT core itself.

Some OT configs actually configure  the implementation of `OT_ASSERT()` to use the `otPlatAssertFail()` provided by platform, so we rely on platform layer to provide the assert functionality for core (won't make sense to then allow platform to use the core definition).

Keeping it separate also allows  project integrators be selective about where to enable/disable asserts (e.g., disable assert in OT core but keep it enabled at platform layer or in some specific driver).

We can consider treating assert functionality similar to `common/code_utils.hpp` which is for core only, but then we added similar helper macros in `example/utils/code_utils.h` for platform layers to use.

Reply:
Make sense. Thanks for the explanation. @abtink